### PR TITLE
execute: log a warning when setting SELinux context fails

### PR DESCRIPTION
**Pull request by topimiettinen**
Update also manual page to explain how the transition can still fail.

### CODE REVIEW

File: man/systemd.exec.xml [hunk 1]
```diff
@@ -730,10 +730,13 @@ CapabilityBoundingSet=~CAP_B CAP_C</programlisting>
 
         <listitem><para>Set the SELinux security context of the executed process. If set, this will override the
         automated domain transition. However, the policy still needs to authorize the transition. This directive is
-        ignored if SELinux is disabled. If prefixed by <literal>-</literal>, all errors will be ignored. This does not
-        affect commands prefixed with <literal>+</literal>.  See <citerefentry
-        project='die-net'><refentrytitle>setexeccon</refentrytitle><manvolnum>3</manvolnum></citerefentry> for
-        details.</para></listitem>
+        ignored if SELinux is disabled. If prefixed by <literal>-</literal>, failing to set the SELinux
+        security context will be ignored, but it's still possible that the subsequent
+        <function>execve()</function> may still fail if the policy doesn't allow the transition for the
```
Comment (yuwata) [inline man/systemd.exec.xml]: "still" is doubled. I think "may still" should be dropped.

**Review by yuwata — APPROVED** [review]
LGTM. But one minor suggestion.

**Comment by akontsevoy** [comment]
Would it not be better to drop the mention of `-` prefix from documentation entirely? It's not been working at all ut until recently, until #21189 only partially fixed it, but even so, the way it may break is non-obvious to someone unfamiliar with the details of selinux.

File: man/systemd.exec.xml (hunk 1 above)
Comment (topimiettinen) [reply man/systemd.exec.xml]: Thanks, updated.

**Comment by topimiettinen** [comment]
> Would it not be better to drop the mention of `-` prefix from documentation entirely? It's not been working at all until recently, until #21189 only partially fixed it, but even so, the way it may break is non-obvious to someone unfamiliar with the details of selinux. To me, all this implies that "it should not be used in production."

Well, SELinux policies can work or break in very non-obvious ways for even someone more familiar with the details. The problem could be also in the further steps after `execve()`, for example the application might not be able to load any libraries. Maybe there are already policies which work correctly as designed when the context is applied and also when it isn't, for example along the lines of your earlier example. I think with the updated manual page and logging we clearly put the responsibility to the policy writer. If I used this method of setting the type, I'd verify that the launched process indeed gets the correct label and if it didn't, I'd check the logs for clues or perhaps even read the fine manual.

**Comment by poettering** [comment]
> Would it not be better to drop the mention of `-` prefix from documentation entirely? It's not been working at all until recently, until #21189 only partially fixed it, but even so, the way it may break is non-obvious to someone unfamiliar with the details of selinux. To me, all this implies that "it should not be used in production."

Dunno, "-" means "all bets are off, we'll continue on any selinux failure, bad things might happen". If you don't want to accept that behaviour, then simply don't use it? It's expressly opt-in after all?

**Comment by keszybz** [comment]
LGTM too.

**Pull request merged**